src/charm.py

```diff
@@ -7,48 +7,63 @@
 import logging
 from typing import Any
 
+import ops
 import requests
 from charms.data_platform_libs.v0.data_interfaces import DatabaseCreatedEvent
 from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires
-from ops.charm import CharmBase
-from ops.main import main
-from ops.model import ActiveStatus
-from ops.model import BlockedStatus
-from ops.model import MaintenanceStatus
-from ops.model import WaitingStatus
-from ops.pebble import Layer
 
 # Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
 PEER_NAME = "fastapi-peer"
 
 
-class FastAPIDemoCharm(CharmBase):
+class FastAPIDemoCharm(ops.CharmBase):
     """Charm the service."""
 
-    def __init__(self, *args):
-        super().__init__(*args)
+    def __init__(self, framework):
+        super().__init__(framework)
         self.pebble_service_name = "fastapi-service"
-        self.container = self.unit.get_container("demo-server")  # see 'containers' in metadata.yaml
+        self.container = self.unit.get_container(
+            "demo-server"
+        )  # see 'containers' in charmcraft.yaml
 
         # Charm events defined in the database requires charm library.
         self.database = DatabaseRequires(self, relation_name="database", database_name="names_db")
-        self.framework.observe(self.database.on.database_created, self._on_database_created)
-        self.framework.observe(self.database.on.endpoints_changed, self._on_database_created)
-        self.framework.observe(self.on.database_relation_broken, self._on_database_relation_removed)
-
-        self.framework.observe(self.on.demo_server_pebble_ready, self._update_layer_and_restart)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-
-        self.framework.observe(self.on.start, self._count)
+        framework.observe(self.database.on.database_created, self._on_database_created)
+        framework.observe(self.database.on.endpoints_changed, self._on_database_created)
+
+        framework.observe(self.on.demo_server_pebble_ready, self._update_layer_and_restart)
+        framework.observe(self.on.config_changed, self._on_config_changed)
+        framework.observe(self.on.collect_unit_status, self._on_collect_status)
+        framework.observe(self.on.start, self._count)
+
+    def _on_collect_status(self, event):
```

Comment:
I think this function contains too many functionalities and is too complicated, but since it's already live in the juju sdk docs, maybe we can leave it like this for now?